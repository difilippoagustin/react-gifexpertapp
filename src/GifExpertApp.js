import React, { useState } from "react";
import { AddCategorie } from "./components/AddCategorie";
import GifGrid from "./components/GifGrid";
export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Simpsons"]);

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
      <AddCategorie setCategories={setCategories} />
      <ul>
        {categories.map((category) => {
          return <GifGrid key={category} category={category} />;
        })}
      </ul>
    </>
  );
};
// AoFoTkxKPHX3mXe9F5iI9sC5Ynv7cfZz
