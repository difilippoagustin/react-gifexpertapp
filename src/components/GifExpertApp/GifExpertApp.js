import React, { useState } from "react";
import { AddCategorie } from "../AddCategorie/AddCategorie";
import GifGrid from "../GifGrid/GifGrid";
export const GifExpertApp = ({ defaultCategories = [] }) => {
  // const [categories, setCategories] = useState(["Simpsons"]);
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
      <AddCategorie setCategories={setCategories} />
      <ul>
        {categories.map((category) => {
          return <GifGrid key={category} category={category} />;
        })}
      </ul>
    </>
  );
};
