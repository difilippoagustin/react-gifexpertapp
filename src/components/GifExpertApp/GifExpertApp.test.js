import React from "react";
import "@testing-library/jest-dom";

import { GifExpertApp } from "./GifExpertApp";
import { shallow } from "enzyme";

describe("<GifExpertApp />", () => {
  test("should display correctly", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });
  test("should display a categories list", () => {
    const mockCategories = ["Simpsons", "Futurama"];
    const wrapper = shallow(
      <GifExpertApp defaultCategories={mockCategories} />
    );
    expect(wrapper.find("GifGrid").length).toBe(2);
  });
});
