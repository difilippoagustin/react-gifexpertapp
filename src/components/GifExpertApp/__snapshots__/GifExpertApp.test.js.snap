// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp /> should display correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategorie
    setCategories={[Function]}
  />
  <ul />
</Fragment>
`;
